### DISCUSSION TITLE
add user/account/profile in redirect callback params

### DISCUSSION THREAD

**Discussion by Feeco-Li** [nextauthjs/next-auth#11269] [Ideas]
### Goals

1. I want to redirect user to specific page (/admin vs /client) based on user's role after login.

### Non-Goals

1.
2.
3.

### Background

when i configure redirect callback in the next-auth config, I found that only 2 params are passed into, "url" and "baseUrl"

### Proposal

I don't know, I'm fairly new to next-auth/auth lib.

**Comment by k-taro56** [comment]
When can we get a user role?
Before sign-in is complete? Or after sign-in is complete?

**Reply by Feeco-Li** [reply]
its based on your "POST sign in " reponse.

**Reply by Feeco-Li** [reply]
> When can we get a user role? Before sign-in is complete? Or after sign-in is complete?

oh wait, you're a maintainer of next-auth, right? 

so, sign-in callback support return "/redirect-url", however, once I redirect by user object fetched from database, the subsequent callbacks won't execute (jwt/session). so, still i have to return boolean in sign-in callback.

user object is passed from sign-in callback to jwt/session callback. why not redirect. that's my suggestion.
